osu.Game/Rulesets/Difficulty/PerformanceBreakdownCalculator.cs

```diff
@@ -36,11 +38,53 @@ public async Task<PerformanceBreakdown> CalculateAsync(ScoreInfo score, Cancella
             PerformanceAttributes[] performanceArray = await Task.WhenAll(
                 // compute actual performance
                 performanceCache.CalculatePerformanceAsync(score, cancellationToken),
+                // compute performance for a full combo
+                getFCPerformance(score, cancellationToken),
                 // compute performance for perfect play
                 getPerfectPerformance(score, cancellationToken)
             ).ConfigureAwait(false);
 
-            return new PerformanceBreakdown(performanceArray[0] ?? new PerformanceAttributes(), performanceArray[1] ?? new PerformanceAttributes());
+            return new PerformanceBreakdown(performanceArray[0] ?? new PerformanceAttributes(), performanceArray[1] ?? new PerformanceAttributes(), performanceArray[2] ?? new PerformanceAttributes());
+        }
+
+        [ItemCanBeNull]
+        private Task<PerformanceAttributes> getFCPerformance(ScoreInfo score, CancellationToken cancellationToken = default)
+        {
+            return Task.Run(async () =>
+            {
+                Ruleset rulest = score.Ruleset.CreateInstance();
+                ScoreInfo fcPlay = score.DeepClone();
+                fcPlay.Passed = true;
+                // Update the play to be a full combo
+                fcPlay.MaxCombo = calculateMaxCombo(playableBeatmap);
+
+                // Only recalculate accuracy if it's different
+                if (fcPlay.Statistics[HitResult.Miss] > 0)
+                {
+                    // number of hits which are worth 300 or more score (greats and perfects)
+                    int count300 = fcPlay.Statistics.Where(x => Judgement.ToNumericResult(x.Key) >= 300).Sum(x => x.Value);
+                    // The ratio of 300s to other values not including misses
+                    double ratio300 = (double)count300 / (fcPlay.Statistics.Sum(x => x.Value) - fcPlay.Statistics[HitResult.Miss]);
+                    // The number of extra greats to add
+                    int newGreat = (int)Math.Round(fcPlay.Statistics[HitResult.Miss] * ratio300);
+                    // The number of extra oks to add
+                    int newOk = fcPlay.Statistics[HitResult.Miss] - newGreat;
+                    fcPlay.Statistics[HitResult.Great] += newGreat;
+                    fcPlay.Statistics[HitResult.Ok] += newOk;
+                    fcPlay.Statistics[HitResult.Miss] = 0;
+                    // Recalculate Accuracy with misses converted to 300s
+                    fcPlay.CalculateAccuracy();
+                }
```

Comment:
The above concerns notwithstanding, this is a complete no-go though. This is just 100% assuming osu! and not even attempting to account for other rulesets in any way.